sys/shell/commands/sc_openwsn.c

```diff
@@ -0,0 +1,578 @@
+#include "cjoin.h"
+#endif
+
+extern idmanager_vars_t idmanager_vars;
+extern neighbors_vars_t neighbors_vars;
+extern openqueue_vars_t openqueue_vars;
+extern schedule_vars_t schedule_vars;
+extern scheduler_dbg_t scheduler_dbg;
+
+#ifdef MODULE_OPENWSN_IPV6
+extern icmpv6rpl_vars_t icmpv6rpl_vars;
+#endif
+
+char addr_str[IPV6_ADDR_MAX_STR_LEN];
+
+static const struct {
+    char *name;
+    int id;
+} components[] = {
+    { "?", COMPONENT_NULL },
+    { "openwsn", COMPONENT_OPENWSN },
+    { "idmanager", COMPONENT_IDMANAGER },
+    { "openqueue", COMPONENT_OPENQUEUE },
+    { "openserial", COMPONENT_OPENSERIAL },
+    { "pktfuncs", COMPONENT_PACKETFUNCTIONS },
+    { "random", COMPONENT_RANDOM },
+    { "radio", COMPONENT_RADIO },
+    { "154", COMPONENT_IEEE802154 },
+    { "154e", COMPONENT_IEEE802154E },
+    { "6top2154e", COMPONENT_SIXTOP_TO_IEEE802154E },
+    { "154e26top", COMPONENT_IEEE802154E_TO_SIXTOP },
+    { "6top", COMPONENT_SIXTOP },
+    { "neigh", COMPONENT_NEIGHBORS },
+    { "sched", COMPONENT_SCHEDULE },
+    { "6topres", COMPONENT_SIXTOP_RES },
+    { "bridge", COMPONENT_OPENBRIDGE },
+    { "iphc", COMPONENT_IPHC },
+    { "frag", COMPONENT_FRAG },
+    { "fwd", COMPONENT_FORWARDING },
+    { "icmpv6", COMPONENT_ICMPv6 },
+    { "icmpv6echo", COMPONENT_ICMPv6ECHO },
+    { "icmpv6router", COMPONENT_ICMPv6ROUTER },
+    { "icmpv6rpl", COMPONENT_ICMPv6RPL },
+    { "udp", COMPONENT_OPENUDP },
+    { "coap", COMPONENT_OPENCOAP },
+    { "cjoin", COMPONENT_CJOIN },
+    { "oscore", COMPONENT_OSCORE },
+    { "c6t", COMPONENT_C6T },
+    { "uinject", COMPONENT_UINJECT },
+};
+
+char *_get_component(int id)
+{
+    for (unsigned i = 0; i < ARRAY_SIZE(components); i++) {
+        if (id == components[i].id) {
+            return components[i].name;
+        }
+    }
+    return NULL;
+}
+
+int _openwsn_ifconfig(char *arg)
+{
+    (void)arg;
+
+    open_addr_t *hwaddr;
+    open_addr_t temp_my128bID;
+
+    memcpy(&temp_my128bID.addr_128b[0], &idmanager_vars.myPrefix.prefix, 8);
+    memcpy(&temp_my128bID.addr_128b[8], &idmanager_vars.my64bID.addr_64b, 8);
+
+    printf("Iface  %d  ", openwsn_get_pid());
+    hwaddr = idmanager_getMyID(ADDR_16B);
+    printf("\tHWaddr: %s  ", netif_addr_to_str(hwaddr->addr_16b, 2, addr_str));
+    hwaddr = idmanager_getMyID(ADDR_PANID);
+    printf("NID: %s\n", netif_addr_to_str(hwaddr->panid, 2, addr_str));
+    printf("\n");
+
+    hwaddr = idmanager_getMyID(ADDR_64B);
+    printf("\t\tLong HWaddr: %s\n",
+           netif_addr_to_str(hwaddr->addr_64b, 8, addr_str));
+
+#ifdef MODULE_OPENWSN_IPV6
+    ipv6_addr_to_str(addr_str, (ipv6_addr_t *)temp_my128bID.addr_128b,
+                     sizeof(addr_str));
+    printf("\t\tinet6 addr: %s\n", addr_str);
+    printf("\n");
+#endif
+
+    printf("\t\tIEEE802154E sync: %i\n", ieee154e_isSynch());
+#ifdef MODULE_OPENWSN_CJOIN
+    printf("\t\t6TiSCH joined: %i\n", cjoin_getIsJoined());
+#endif
+    printf("\n");
+
+#ifdef MODULE_OPENWSN_IPV6
```

Comment:
You are using the `IS_USED` below without preprocessor conditionals. Is it possible to it here as well ?